### PR TITLE
Multiple CIs in table

### PR DESCRIPTION
**Pull request by DominiqueMakowski**
to handle this type of thing:

``` r
x <- structure(list(Parameter = c("Species", "Cat1", "Species:Cat1", "Residuals"), Sum_Squares = c(63.2121333333334, 0.00166666666666666, 0.524133333333333, 38.4304), df = c(2, 1, 2, 144), Mean_Square = c(31.6060666666667, 0.00166666666666666, 0.262066666666666, 0.266877777777778), F = c(118.42899371331, 0.00624505599733542, 0.981972605021023, NA), p = c(3.86508033747795e-31, 0.937121919138614, 0.377065429569501, NA), Eta2_partial = c(0.621906314810467, 4.33665636855333e-05, 0.0134550022419299, NA), Eta2_partial_CI_low = c(0.544281560034728, 0, 0, NA), Eta2_partial_CI_high = c(0.681676982792917, 0.00317952175831753, 0.0512263475038747, NA)), row.names = c(NA, -4L), class = "data.frame")
parameters::parameters_table(x)
#>      Parameter Sum_Squares  df Mean_Square        F      p Eta2 (partial)
#> 1      Species       63.21   2       31.61   118.43 < .001           0.62
#> 2         Cat1    1.67e-03   1    1.67e-03 6.25e-03 0.937        4.34e-05
#> 3 Species:Cat1        0.52   2        0.26     0.98 0.377            0.01
#> 4    Residuals       38.43 144        0.27                               
#>             CI
#> 1 [0.54, 0.68]
#> 2 [0.00, 0.00]
#> 3 [0.00, 0.05]
#> 4
```

<sup>Created on 2020-10-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

but we're not there yet

### CODE REVIEW

**Comment by DominiqueMakowski** [comment]
``` r
# With the latest dev report
x <- aov(Sepal.Length ~ Species, data=iris)
report::report_table(x)
#> Parameter | Sum_Squares |  df | Mean_Square |      F |      p | Eta2 (partial) |       90% CI
#> ---------------------------------------------------------------------------------------------
#> Residuals |       38.96 | 147 |        0.27 |        |        |                |             
#> Species   |       63.21 |   2 |       31.61 | 119.26 | < .001 |           0.62 | [0.54, 0.68]
```

<sup>Created on 2020-10-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Seems to work @strengejacke you're okay to merge? It should not break anything else?

**Comment by IndrajeetPatil** [comment]
This partially addresses https://github.com/easystats/parameters/issues/251, right?

**Comment by DominiqueMakowski** [comment]
If we include the CIs and rename them in the model_parameters table, then yes in theory at least it should be printed correctly ^^

**Comment by DominiqueMakowski** [comment]
Also, we have to see if we want to print a barebone "...%CI" or if you want to add in the name of the effect size index

File: R/parameters_table.R [hunk 1]
```diff
@@ -78,6 +78,32 @@ parameters_table <- function(x, pretty_names = TRUE, stars = FALSE, digits = 2,
     x <- x[!names(x) %in% c(ci_low, ci_high)]
   }
 
+  # Other CIs
+  other_ci_low <- names(x)[grep("_CI_low", names(x))]
```
Comment (strengejacke) [inline R/parameters_table.R:82]: If we can be sure (and make this a naming convention) that "other" CIs follow the pattern `<my_parameter_name>_CI_low`, you could add a `$` to the pattern, to ensure it only captures `_CI_low` and `_CI_high` at the end of a string;

```r
other_ci_low <- names(x)[grep("_CI_low$", names(x))]
```

same for `strsplit()`.

Comment (DominiqueMakowski) [reply R/parameters_table.R:82]: > hat "other" CIs follow the pattern

Well, we can decide on that convention. As far as I know its first usage will be in this context of estimate + effect size

Comment (strengejacke) [reply R/parameters_table.R:82]: @mattansb ?

Comment (mattansb) [reply R/parameters_table.R:86]: What is and is not captured by "other"?

File: R/parameters_table.R [hunk 2]
```diff
@@ -59,14 +59,18 @@ parameters_table <- function(x, pretty_names = TRUE, stars = FALSE, digits = 2,
     x$p <- format(x$p, justify = "left")
   }
 
-  # CI
+  # Main CI
   ci_low <- names(x)[grep("CI_low*", names(x))]
```
Comment (strengejacke) [inline R/parameters_table.R:63]: I think this pattern should be 

```r
grep("^CI_low", names(x))
```

and the others respectively, else `"Eta2_partial_CI_low"` is already caught here.

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/parameters_table.R (hunk 1 above)
Comment (strengejacke) [reply R/parameters_table.R:86]: > What is and is not captured by "other"?

The intention is that all CI-columns _starting_ with `CI_low` or `CI_high` are processed by default. Then, all other CI columns that _end_ with `_CI_low` or `_CI_high` (like `Eta2_partial_CI_low`) are also processed (captured by "other").

**Comment by strengejacke** [comment]
When I look at the code, I'm not sure what it should do, as it didn't change any current behaviour, or?

**Comment by DominiqueMakowski** [comment]
> I'm not sure what it should do

you can merge when you approve :)

**Comment by DominiqueMakowski** [comment]
No wait it's still not perfect. Best to wait until I've finished having a complete overview of the cases in report

File: R/parameters_table.R (hunk 1 above)
Comment (mattansb) [reply R/parameters_table.R:86]: Sounds okay to me 👍

File: R/parameters_table.R [hunk 3]
```diff
@@ -60,16 +60,16 @@ parameters_table <- function(x, pretty_names = TRUE, stars = FALSE, digits = 2,
   }
 
   # Main CI
-  ci_low <- names(x)[grep("CI_low*", names(x))]
-  ci_high <- names(x)[grep("CI_high*", names(x))]
+  ci_low <- names(x)[grep("^CI_low.*", names(x))]
```
Comment (strengejacke) [inline R/parameters_table.R:63]: I don't think you need the quantifier here, `"^CI_low"` should be enough.

File: R/parameters_table.R [hunk 4]
```diff
@@ -60,16 +60,16 @@ parameters_table <- function(x, pretty_names = TRUE, stars = FALSE, digits = 2,
...
+  ci_high <- names(x)[grep("^CI_high.*", names(x))]
   if (length(ci_low) >= 1 & length(ci_low) == length(ci_high)) {
     if (is.null(attributes(x)$ci)) {
       ci_colname <- "CI"
     } else {
-      if length(unique(attributes(x)$ci) > 1){
+      if (length(unique(na.omit(attributes(x)$ci))) > 1){
```
Comment (strengejacke) [inline R/parameters_table.R:69]: please use `stats::na.omit` :-) and don't forget whitespace.

**Review by strengejacke — CHANGES_REQUESTED** [review]

**Comment by DominiqueMakowski** [comment]
This function has become quite messy and would benefit from a cleaning maybe by splitting the different operations in different internals :)

**Comment by strengejacke** [comment]
@DominiqueMakowski Now please provide an example what this PR does new, and maybe a test, so we can merge this.

**Comment by strengejacke** [comment]
Oh, and **update the news file**!

**Comment by DominiqueMakowski** [comment]
> Oh, and update the news file!

There's nothing to update it just makes it work as it should for some future cases, like:

``` r
report::report_table(lm(Sepal.Length ~ Species, data=iris))
#> Parameter         | Coefficient |   SE |       95% CI |     t |  df |      p | Coefficient (std.) | Coefficient 95% CI (std.)
#> -----------------------------------------------------------------------------------------------------------------------------
#> (Intercept)       |        5.01 | 0.07 | [4.86, 5.15] | 68.76 | 147 | < .001 |              -1.01 |            [-1.18, -0.84]
#> Speciesversicolor |        0.93 | 0.10 | [0.73, 1.13] |  9.03 | 147 | < .001 |               1.12 |            [ 0.88,  1.37]
#> Speciesvirginica  |        1.58 | 0.10 | [1.38, 1.79] | 15.37 | 147 | < .001 |               1.91 |            [ 1.66,  2.16]
```

<sup>Created on 2020-10-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
ok, please add some tests, your last implementation broke this code:

```r
m <- lm(Sepal.Length ~ Species, data=iris)
parameters::model_parameters(m, standardize = "refit")
```

**Comment by strengejacke** [comment]
> There's nothing to update it just makes it work as it should for some future cases

Well, that is actually a news ;-) `parameters_table()` can now cope with multiple CI columns or so...

**Comment by strengejacke** [comment]
In other words: if this PR did not change anything, there's no need to merge it. 😬

**Comment by strengejacke** [comment]
Also, do you have a reprex that works out of the box (either with master-report, or w/o report required)?

**Comment by DominiqueMakowski** [comment]
```r
> dput(report::report_table(lm(Sepal.Length ~ Species, data=iris)))

structure(list(Parameter = c("(Intercept)", "Speciesversicolor", 
"Speciesvirginica"), Coefficient = c(5.006, 0.930000000000001, 
1.582), SE = c(0.0728022201948961, 0.102957887170494, 0.102957887170494
), CI_low = c(4.86212582748931, 0.726531193960184, 1.37853119396018
), CI_high = c(5.14987417251068, 1.13346880603982, 1.78546880603982
), t = c(68.7616392274662, 9.03281939401064, 15.3655056788439
), df_error = c(147L, 147L, 147L), p = c(1.13428618636292e-113, 
8.77019424057054e-16, 2.21482134895686e-32), Std_Coefficient = c(-1.01119138320282, 
1.12309870984708, 1.91047543976137), Std_Coefficient_CI_low = c(-1.18493858484482, 
0.877383060860579, 1.66475979077487), Std_Coefficient_CI_high = c(-0.837444181560811, 
1.36881435883358, 2.15619108874787)), row.names = c(NA, 3L), class = c("report_table", 
"data.frame"), ci = 0.95, effsize = structure(c("large (Std. beta = -1.01, 95% CI [-1.18, -0.84])", 
"large (Std. beta = 1.12, 95% CI [0.88, 1.37])", "large (Std. beta = 1.91, 95% CI [1.66, 2.16])"
), class = c("report_effectsize", "character"), table = structure(list(
    Parameter = c("(Intercept)", "Speciesversicolor", "Speciesvirginica"
    ), Std_Coefficient = c(-1.01119138320282, 1.12309870984708, 
    1.91047543976137), Std_Coefficient_CI_low = c(-1.18493858484482, 
    0.877383060860579, 1.66475979077487), Std_Coefficient_CI_high = c(-0.837444181560811, 
    1.36881435883358, 2.15619108874787)), row.names = c(NA, -3L
), class = "data.frame"), interpretation = structure(c("large", 
"large", "large"), rule_name = "cohen1988", class = c("effectsize_interpret", 
"character")), statistics = c("Std. beta = -1.01, 95% CI [-1.18, -0.84]", 
"Std. beta = 1.12, 95% CI [0.88, 1.37]", "Std. beta = 1.91, 95% CI [1.66, 2.16]"
), rules = "Effect sizes were labelled following Cohen's (1988) recommendations.", ci = structure(c(0.95, 
0.95, 0.95), .Names = c("ci_Std_Coefficient", NA, NA)), main = c("Std. beta = -1.01", 
"Std. beta = 1.12", "Std. beta = 1.91"), prefix = "  - ", summary = structure(c("large (Std. beta = -1.01, 95% CI [-1.18, -0.84])", 
"large (Std. beta = 1.12, 95% CI [0.88, 1.37])", "large (Std. beta = 1.91, 95% CI [1.66, 2.16])"
), class = c("report_effectsize", "character"), prefix = "  - ")), summary = structure(list(
    Parameter = c("(Intercept)", "Speciesversicolor", "Speciesvirginica"
    ), Coefficient = c(5.006, 0.930000000000001, 1.582), SE = c(0.0728022201948961, 
    0.102957887170494, 0.102957887170494), CI_low = c(4.86212582748931, 
    0.726531193960184, 1.37853119396018), CI_high = c(5.14987417251068, 
    1.13346880603982, 1.78546880603982), t = c(68.7616392274662, 
    9.03281939401064, 15.3655056788439), df_error = c(147L, 147L, 
    147L), p = c(1.13428618636292e-113, 8.77019424057054e-16, 
    2.21482134895686e-32), Std_Coefficient = c(-1.01119138320282, 
    1.12309870984708, 1.91047543976137)), row.names = c(NA, -3L
), class = c("report_table", "data.frame")), ci_Std_Coefficient = structure(c(0.95, 
0.95, 0.95), .Names = c("ci_Std_Coefficient", NA, NA)))
```

**Comment by strengejacke** [comment]
``` r
parameters::parameters_table(dat)
#>           Parameter Coefficient   SE       95% CI     t  df      p
#> 1       (Intercept)        5.01 0.07 [4.86, 5.15] 68.76 147 < .001
#> 2 Speciesversicolor        0.93 0.10 [0.73, 1.13]  9.03 147 < .001
#> 3  Speciesvirginica        1.58 0.10 [1.38, 1.79] 15.37 147 < .001
#>   Coefficient (std.)  (std.) 95% CI
#> 1              -1.01 [-1.18, -0.84]
#> 2               1.12 [ 0.88,  1.37]
#> 3               1.91 [ 1.66,  2.16]
```

<sup>Created on 2020-10-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
or: 

``` r
parameters::parameters_table(dat)
#>           Parameter Coefficient   SE       95% CI     t  df      p Beta (std.)
#> 1       (Intercept)        5.01 0.07 [4.86, 5.15] 68.76 147 < .001       -1.01
#> 2 Speciesversicolor        0.93 0.10 [0.73, 1.13]  9.03 147 < .001        1.12
#> 3  Speciesvirginica        1.58 0.10 [1.38, 1.79] 15.37 147 < .001        1.91
#>   Beta (std.) 95% CI
#> 1     [-1.18, -0.84]
#> 2     [ 0.88,  1.37]
#> 3     [ 1.66,  2.16]
```

<sup>Created on 2020-10-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

File: R/parameters_table.R [hunk 5]
```diff
@@ -224,10 +228,11 @@ parameters_table <- function(x, pretty_names = TRUE, stars = FALSE, digits = 2,
   }
 
   x[std_cols] <- insight::format_value(x[std_cols], digits = digits)
-  names(x)[names(x) == std_cols] <- "Beta (std.)"
+  names(x)[names(x) == std_cols] <- .replace_words(std_cols, "Std_", "Std. ")
 
   if (!is.null(std_cis) && length(std_cis)) {
-    std_cis_replacement <- gsub("^Std_Coefficient(.*)", "Beta\\1", std_cis)
```
Comment (strengejacke) [inline R/parameters_table.R]: I think the coefficient name for standardized is quite long, so I thought using beta here.

Comment (DominiqueMakowski) [reply R/parameters_table.R]: I agree it's long, but then it should be consistent. Another option is to use abreviations (Coef.) or something. But it's not that big of a deal if it's long imo (but it's definitely a nogo to swap the name, especially for some random unrational, implicit, illogical and inconsistent convention - I'll die on that hill if I need to 😁)

Comment (strengejacke) [reply R/parameters_table.R]: Than coef.? If we compare multiple models in one table, column names need to be short.

Comment (DominiqueMakowski) [reply R/parameters_table.R]: Then, `Coef.` is a variable alternative

**Pull request merged**